src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/JsonSerializerContextTests.cs

```diff
@@ -310,6 +310,187 @@ public static IEnumerable<object[]> GetCombiningContextsData()
+                    Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize(json, typeof(Person), context));
+                }
+                else
+                {
+                    person = (Person)JsonSerializer.Deserialize(json, typeof(Person), context);
+                    VerifySourceGenContextModification(json, person);
+                }
+            }
+        }
+
+        private static void VerifySourceGenContextModification(string json, Person person)
+        {
+            JsonTestHelper.AssertJsonEqual(@"{""FIRSTNAME"":""Jane"",""LASTNAME"":""Doe""}", json);
+            Assert.Equal("Jane", person.FirstName);
+            Assert.Equal("Doe", person.LastName);
+        }
+
+        internal class EditedSourceGenContext : SourceGenContext
+        {
+            public override JsonTypeInfo GetTypeInfo(Type type)
+                => EditTypeInfo(base.GetTypeInfo(type));
+        }
+
+        internal class EditedSourceGenContext_Metadata : SourceGenContext_Metadata
+        {
+            public override JsonTypeInfo GetTypeInfo(Type type)
+                => EditTypeInfo(base.GetTypeInfo(type));
+        }
+
+        internal class EditedSourceGenContext_Serialization : SourceGenContext_Serialization
+        {
+            public override JsonTypeInfo GetTypeInfo(Type type)
+                => EditTypeInfo(base.GetTypeInfo(type));
+        }
+
+        internal class EditedSourceGenContext_WithCheck : SourceGenContext
+        {
+            public override JsonTypeInfo GetTypeInfo(Type type)
+                => EditTypeInfo(base.GetTypeInfo(type), checkIfPropsConfigured: true);
+        }
+
+        internal class EditedSourceGenContext_WithCheck_Metadata : SourceGenContext_Metadata
+        {
+            public override JsonTypeInfo GetTypeInfo(Type type)
+                => EditTypeInfo(base.GetTypeInfo(type), checkIfPropsConfigured: true);
+        }
+
+        internal class EditedSourceGenContext_WithCheck_Serialization : SourceGenContext_Serialization
+        {
+            public override JsonTypeInfo GetTypeInfo(Type type)
+                => EditTypeInfo(base.GetTypeInfo(type), checkIfPropsConfigured: true);
+        }
+
+        internal static JsonTypeInfo EditTypeInfo(JsonTypeInfo typeInfo, bool checkIfPropsConfigured = false)
+        {
+            if (typeInfo.Kind == JsonTypeInfoKind.Object)
+            {
+                foreach (JsonPropertyInfo property in typeInfo.Properties)
+                {
+                    // Do we need a public API indicating whether a JsonPropertyInfo is configured?
+                    // If we allow custom `JsonSerializerContext.GetTypeInfo(Type)` implementations,
+                    // then they will need this API or similar since the serializer requests type info from
+                    // them potentially multiple times (i.e repeated serializer calls)
+                    if (checkIfPropsConfigured &&
```

Comment:
Do we need a public API indicating whether a `JsonPropertyInfo` instance is configured? If we allow custom `JsonSerializerContext.GetTypeInfo(Type)` implementations, then they will need this API or similar since the serializer requests type info from them potentially multiple times (i.e repeated serializer calls)